.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,219 @@
+    steps:
+      - uses: actions/checkout@v2
+      - id: dependency-versions-matrix
+        run: |
+          DEPS=$(if [ -f composer.lock ]; then echo -n '["lowest", "locked", "latest"]'; else echo -n '["lowest", "latest"]'; fi)
+          echo "Testing against dependencies: ${DEPS}"
+          echo "::set-output name=check::${DEPS}"
+
+  stable-php:
+    name: Determine Stable PHP Version for Non-Unit Test Checks
+    runs-on: ubuntu-latest
+    needs: [supported-versions-matrix]
+    outputs:
+      version: ${{ steps.stable-php.outputs.version }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: stable-php
+        run: |
+          # This grabs the default stable PHP version for non-unit-test checks from the laminas.dev API.
+          # If the .github/workflows/continuous-integration-config.json file is
+          # present, and defines a string "stable-php" element, that value will
+          # be used.
+          DEFAULT_VERSION=$(curl -s https://laminas.dev/api/ci/stable-checks-version | jq -r '.version')
+          VERSION=$(if [ -f .github/workflows/continuous-integration-config.json ]; then jq -r 'def isstring(x): if x|type=="string" then x else empty end ; isstring(.["stable-php"])' < .github/workflows/continous-integration-config.json ; fi)
+          VERSION=${VERSION:-$DEFAULT_VERSION}
+          echo "PHP version discovered for non-unit-test checks: ${VERSION}"
+          echo "::set-output name=version::${VERSION}"
+
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    needs: [supported-versions-matrix, dependency-versions-matrix]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version: ${{ fromJson(needs.supported-versions-matrix.outputs.version) }}
+        deps-version: ${{ fromJson(needs.dependency-versions-matrix.outputs.version) }}
+        include:
+          - php-version: "8.0"
+            composer-args: "--ansi --no-interaction --no-progress --prefer-dist --ignore-platform-reqs"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          extensions: mbstring, intl, readline
+          ini-values: memory_limit=-1
+          coverage: none
+          tools: composer:v2, cs2pr
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v2"
+        with:
+          path: |
+            ~/.composer/cache
+            vendor
+          key: "php-${{ matrix.php-version }}-${{ matrix.deps-version }}-${{ hashFiles('**/composer.lock') || github.ref }}"
+          restore-keys: "php-${{ matrix.php-version }}-${{ matrix.deps-version }}-${{ hashFiles('**/composer.lock') || github.ref }}"
+
+      - name: "Install dependencies (lowest)"
+        if: matrix.deps-version == 'lowest'
+        env:
+          COMPOSER_ARGS: ${{ matrix.composer-args || '--ansi --no-interaction --no-progress --prefer-dist' }}
```

Comment:
Is there a way to specify these `--ansi --no-interaction --no-progress --prefer-dist` globally? 
Its repeated over and over again here so that would be a thing.

Reply:
I made another step that defines the default and PHP8 composer arguments as outputs, and now this becomes `${{ matrix.composer-args || needs.composer-args.outputs.default }}`. There's still some repetition, but it's easier to catch.